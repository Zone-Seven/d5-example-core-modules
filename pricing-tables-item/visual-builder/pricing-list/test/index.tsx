import React from 'react';
import { shallow } from 'enzyme';

import { PricingList } from '..';


describe('<PricingList />', () => {
  const content = `+This feature is included
  This feature is included
  +This feature is included
  -This feature is included
  -This feature is included`;

  const pricingList = shallow(
    <PricingList
      content={content}
    />,
  );

  it('Should match the snapshot when well formatted parameter is given', () => {
    expect(pricingList.html()).toMatchSnapshot();
  });

  const contentWithP = `<p>+This feature is included</p><p>+This feature is included</p><p>-This feature is included</p>
  <p>-This feature is included</p>`;

  const pricingListWithP = shallow(
    <PricingList
      content={contentWithP}
    />,
  );

  it('Should match the snapshot when content contains <p> tag', () => {
    expect(pricingListWithP.html()).toMatchSnapshot();
  });

  const contentWithDiv = `<div>+This feature is included</div><div>+This feature is included</div><div>-This feature is included</div>
  <div>-This feature is included</div>`;

  const pricingListWithDiv = shallow(
    <PricingList
      content={contentWithDiv}
    />,
  );

  it('Should match the snapshot when content contains <div> tag', () => {
    expect(pricingListWithDiv.html()).toMatchSnapshot();
  });

  const contentAsNumber = 20;

  const pricingListWithNumber = shallow(
    <PricingList
      content={contentAsNumber as unknown as string}
    />,
  );

  it('Should match the snapshot with number content', () => {
    expect(pricingListWithNumber.html()).toMatchSnapshot();
  });

  // eslint-disable-next-line no-undefined
  const contentAsUndefined: unknown = undefined;

  const pricingListWithUndefined = shallow(
    <PricingList
      content={contentAsUndefined as string}
    />,
  );

  it('Should match the snapshot with undefined content', () => {
    expect(pricingListWithUndefined.html()).toMatchSnapshot();
  });
});
