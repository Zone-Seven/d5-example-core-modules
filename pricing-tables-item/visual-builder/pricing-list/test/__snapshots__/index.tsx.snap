// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PricingList /> Should match the snapshot when content contains <div> tag 1`] = `
<ul class="et_pb_pricing">
    <li><span>&lt;div&gt;+This feature is included&lt;/div&gt;&lt;div&gt;+This feature is included&lt;/div&gt;&lt;div&gt;-This feature is included&lt;/div&gt;</span></li>
    <li><span>&lt;div&gt;-This feature is included&lt;/div&gt;</span></li>
</ul>
`;

exports[`<PricingList /> Should match the snapshot when content contains <p> tag 1`] = `
<ul class="et_pb_pricing">
    <li><span>This feature is included</span></li>
    <li><span>This feature is included</span></li>
    <li class="et_pb_not_available"><span>This feature is included</span></li>
    <li class="et_pb_not_available"><span>This feature is included</span></li>
</ul>
`;

exports[`<PricingList /> Should match the snapshot when well formatted parameter is given 1`] = `
<ul class="et_pb_pricing">
    <li><span>This feature is included</span></li>
    <li><span>This feature is included</span></li>
    <li><span>This feature is included</span></li>
    <li class="et_pb_not_available"><span>This feature is included</span></li>
    <li class="et_pb_not_available"><span>This feature is included</span></li>
</ul>
`;

exports[`<PricingList /> Should match the snapshot with number content 1`] = `
<ul class="et_pb_pricing">
    <li><span>20</span></li>
</ul>
`;

exports[`<PricingList /> Should match the snapshot with undefined content 1`] = `<ul class="et_pb_pricing"></ul>`;
